### PR TITLE
orc writer

### PR DESCRIPTION
**Pull request by razajafri**
Writes the cudf table to an ORC file given the path. 

Currently it doesn't support buffer

### CODE REVIEW

**Comment by razajafri** [comment]
rerun tests

**Comment by razajafri** [comment]
@revans2 @jlowe I have added more tests PTAL. I want you to notice that I have added a package-private method for replacing columns

File: java/src/main/java/ai/rapids/cudf/ORCWriterOptions.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    compressionType = builder.compressionType;
+  }
+
+  public CompressionType getCompressionType() {
+    return compressionType;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/ORCWriterOptions.java]: Style nit: need whitespace after this line.

File: java/src/main/java/ai/rapids/cudf/ORCWriterOptions.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  public static class Builder {
+    private CompressionType compressionType = CompressionType.SNAPPY;
+
+    /**
+     * Specify the compression type to use with this file
+     * @return
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/ORCWriterOptions.java]: Javadoc directive with no text.  Either this should not be a Javadoc-style comment or there needs to be text explaining what is being returned.  It's also a bit inconsistent to have Javadoc comments for this method but not other public methods within the class.

**Pull request merged**